samples/sample_decode/src/pipeline_decode.cpp

```diff
@@ -389,8 +389,9 @@ mfxStatus CDecodingPipeline::Init(sInputParams *pParams)
         m_bVppFullColorRange = pParams->bUseFullColorRange;
     }
 
-    m_bVppIsUsed = IsVppRequired(pParams);
-    if (m_bVppIsUsed)
+    bool bResolutionSpecified = pParams->Width || pParams->Height; // potentially VPP can be inserted
+
+    if (bResolutionSpecified)
         m_bDecOutSysmem = pParams->bUseHWLib ? false : true;
```

Comment:
This is a bit different from what I proposed. If we can't rely on IsVppRequired here, I'm not sure we can set m_bDecOutSysmem here.  I don't say it's wrong, just not sure.